torch/csrc/distributed/autograd/context/dist_autograd_container.h

```diff
@@ -104,7 +111,19 @@ class TORCH_API DistAutogradContainer {
   int16_t worker_id_;
 
   // Map from autograd_context_id to DistAutogradContext.
-  std::unordered_map<int64_t, DistAutogradContext> autograd_context_;
+  std::unordered_map<int64_t, std::tuple<DistAutogradContext, std::chrono::time_point<std::chrono::high_resolution_clock>>> autograd_context_;
```

Comment:
Create a small struct `DistAutogradContextWithTimeout` instead of using a tuple like this.